web/app/app.go

```diff
@@ -201,29 +201,50 @@ func Run(defaultHost string, routes http.Handler, readTimeout, writeTimeout time
 	}
 
 	// Create a new server and set timeout values.
-	s := manners.NewWithServer(&http.Server{
+	server := manners.NewWithServer(&http.Server{
 		Addr:           host,
 		Handler:        routes,
 		ReadTimeout:    readTimeout,
 		WriteTimeout:   writeTimeout,
 		MaxHeaderBytes: 1 << 20,
 	})
 
-	// Support for shutting down cleanly.
+	serverErrors := make(chan error)
+	osSignals := make(chan os.Signal)
+
+	// Start the server listening inside a seperate go-routine so we can capture
+	// any error that occurs.
 	go func() {
+		log.Dev("listener", "Run", "Listening on: %s", server.Addr)
+		if err := server.ListenAndServe(); err != nil {
+			serverErrors <- err
+		}
+	}()
 
-		// Listen for an interrupt signal from the OS.
-		sigChan := make(chan os.Signal, 1)
-		signal.Notify(sigChan, os.Interrupt)
-		<-sigChan
+	// Listen for an interrupt signal from the OS.
+	signal.Notify(osSignals, os.Interrupt)
 
-		// We have been asked to shutdown the server.
-		log.Dev("shutdown", "Run", "Starting shutdown...")
-		s.Close()
-	}()
+	// Block until the interrupt signal is recieved.
+	for {
+		select {
+		case err := <-serverErrors:
+			if err != nil {
+				log.Fatal("shutdown", "Run", "Error Occured: %s", err.Error())
```

Comment:
This could be changed if we accepted that calling "Run" would return an error instead? Fatal was chosen as general use case had it that the sever "which was a web server" should shut down (perhaps with an error) if the web server could not start (or has an issue starting).